src/management/projects.ts

```diff
@@ -13,4 +20,73 @@ export interface DefaultProjectResources {
   locationId: string;
 }
 
-// TODO(caot): Add API methods related to project management into this file
+export enum ProjectParentResourceType {
+  ORGANIZATION = "organization",
+  FOLDER = "folder",
+}
+
+export interface ProjectParentResource {
+  id: string;
+  type: ProjectParentResourceType;
+}
+
+/**
+ * Send an API request to create a new Google Cloud Platform project and poll the LRO to get the
+ * new project information.
+ * @return a promise that resolves to the new cloud project information
+ */
+export async function createCloudProject(
+  projectId: string,
+  options: { displayName?: string; parentResource?: ProjectParentResource }
+): Promise<any> {
+  try {
+    const response = await api.request("POST", "/v1/projects", {
+      auth: true,
+      origin: api.resourceManagerOrigin,
+      timeout: CREATE_PROJECT_API_REQUEST_TIMEOUT_MILLIS,
+      data: { projectId, name: options.displayName || projectId, parent: options.parentResource },
+    });
+
+    const projectInfo = await pollOperation<any>({
+      pollerName: "Project Creation Poller",
+      apiOrigin: api.resourceManagerOrigin,
+      apiVersion: "v1",
+      operationResourceName: response.body.name /* LRO resource name */,
+    });
+    return projectInfo;
+  } catch (err) {
+    logger.debug(err.message);
+    throw new FirebaseError(
+      "Failed to create Google Cloud project. See firebase-debug.log for more info.",
```

Comment:
I don't know that `firebase-debug.log` is _always_ created. Saying to look in that file might cause headache if it doesn't exist. I don't recall the logic for it off the top of my head, but this did catch my eye...

Reply:
I am not 100% sure. Did some reading and I think `firebase-debug.log` is created/appended if the command throws error, which prints all messages we passed into `logger.debug(..)`